### PR TITLE
feat: Add `hideCalendarEventDetails` option to event types

### LINKED ISSUE #16815: [CAL-4410] Add option to make calendar events private on GCal
When creating events on GCal users want the option to make the event private to other users of a shared calendar.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4410](https://linear.app/calcom/issue/CAL-4410/add-option-to-make-calendar-events-private-on-gcal)</sub>

### PR DESCRIPTION
**Pull request by joeauyeung**
Co-authored-by: Alex van Andel <emrysal@users.noreply.github.com>

## What does this PR do?

This PR adds the option to event types to make the created calendar events on 3rd party calendars "private". This definition is different based on the calendar provider.

This functionality is enabled on the following calendars:
- Google calendar
- Outlook
- iCal / Apple calendar
- Generated ICS files in emails

This PR also adds "Don't show other attendees" to Outlook.

- Fixes #16815 (GitHub issue number)
- Fixes CAL-4410 (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- For each calendar integration in this PR
	- Enable "Hide calendar event details" under the advanced tab in an event type
	- Create a booking
	- The 3rd party calendar event should be set as private as defined by the 3rd party integration
	- Disable this option in the event type
	- The 3rd party calendar event should be set at it's default visability

### CODE REVIEW

File: apps/web/lib/event-types/[type]/getServerSideProps.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
     await ssr.viewer.eventTypes.get.prefetch({ id: eventTypeId });
     try {
       const { eventType } = await ssr.viewer.eventTypes.get.fetch({ id: eventTypeId });
+      console.log("🚀 ~ getEventTypeById ~ eventType:", eventType);
```
Comment (keithwillcode) [inline apps/web/lib/event-types/[type]/getServerSideProps.ts:43]: Remove?

File: apps/web/public/static/locales/en/common.json [hunk 2]
```diff
@@ -2613,5 +2613,7 @@
   "limit_team_booking_frequency_description": "Limit how many times members can be booked across all team event types (collective and round-robin)",
   "booking_limits_updated_successfully": "Booking limits updated successfully",
   "you_are_unauthorized_to_make_this_change_to_the_booking": "You are unauthorized to make this change to the booking",
+  "hide_calendar_event_details": "Hide calendar event details",
+  "description_hide_calendar_event_details": "Depending on your calendar, will set calendar event details to private",
```
Comment (keithwillcode) [inline apps/web/public/static/locales/en/common.json:2617]: This is pretty non-descriptive. Where do we plan to tell them which calendars this is supported on?

File: packages/lib/CalendarService.ts [hunk 3]
```diff
@@ -162,6 +162,7 @@ export default abstract class BaseCalendarService implements Calendar {
          * [UPDATE]: Since we're not using the PUBLISH method to publish the iCalendar event and creating the event directly on iCal,
          * this shouldn't be an issue and we should be able to add attendees to the event right here.
          */
+        ...(calEventRaw.hideCalendarEventDetails ? { classification: "PRIVATE" } : {}),
```
Comment (SomayChauhan) [inline packages/lib/CalendarService.ts:165]: this is throwing an error
Screenshot 2024-09-26 at 11 18 22 AM

**Review by ibex088 — DISMISSED** [review]

File: apps/web/public/static/locales/en/common.json (hunk 2 above)
Comment (emrysal) [reply apps/web/public/static/locales/en/common.json:2617]: Agree this isn't the best approach, right now we will support ICS, iCalendar, Apple Calendar, Office 365 Calendar, Outlook Calendar, CalDav and Google Calendar. 

With that range of calendars the question is raised whether we want to mention "limited" calendar support at all.

File: apps/web/public/static/locales/en/common.json [hunk 4]
```diff
@@ -2613,5 +2613,7 @@
   "limit_team_booking_frequency_description": "Limit how many times members can be booked across all team event types (collective and round-robin)",
   "booking_limits_updated_successfully": "Booking limits updated successfully",
   "you_are_unauthorized_to_make_this_change_to_the_booking": "You are unauthorized to make this change to the booking",
+  "hide_calendar_event_details": "Hide calendar event details",
```
Comment (CarinaWolli) [inline apps/web/public/static/locales/en/common.json:2616]: Can we try to make this more descriptive as well?
'Make calendar event private'
'Hide calendar event details on shared calendars'

File: packages/features/bookings/lib/handleNewBooking.ts [hunk 5]
```diff
@@ -898,6 +898,7 @@ async function handler(
     conferenceCredentialId,
     destinationCalendar,
     hideCalendarNotes: eventType.hideCalendarNotes,
+    hideCalendarEventDetails: eventType.hideCalendarEventDetails,
```
Comment (CarinaWolli) [inline packages/features/bookings/lib/handleNewBooking.ts:901]: We need to set `hideCalendarEventDetails` in other places as well. I'd say everywhere where we do `eventManager.create(evt)`, for example in `handleConfirmation`

**Review by CarinaWolli — CHANGES_REQUESTED** [review]
It's not yet working for bookings that are not confirmed by default

**Review by CarinaWolli — DISMISSED** [review]
As discussed with @emrysal, bookings that are not confirmed by default will be handled in a follow-up

**Review by emrysal — DISMISSED** [review]
Happy with platform-libraries bump 👍

**Review by emrysal — APPROVED** [review]

**Pull request merged**